### PR TITLE
Fix enumerating HoveredDrawables causing excessive allocations

### PR DESCRIPTION
**Pull request by peppy**
Screenshots should only be compared using relative % (run times were different):

Before:
20210226 163606 (Safari app)

After:
20210226 163358 (Parallels Desktop app)

### CODE REVIEW

**Comment by smoogipoo** [comment]
Where are the rest of the allocations coming from (the backtraces)? I'd expect `SROLW.GetEnumerator()` to be 0 o_0.

**Comment by peppy** [comment]
I don't think you can drill down into the enumerator itself..

**Comment by smoogipoo** [comment]
Yeah but what's calling GetEnumerator() there?

**Comment by peppy** [comment]
It's the `MenuCursorContainer`, same scenario I was fixing this for.

File: osu.Framework/Input/InputManager.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ public Drawable DraggedDrawable
         /// Top-down in this case means reverse draw order, i.e. the front-most visible
         /// <see cref="Drawable"/> first, and <see cref="Container"/>s after their children.
         /// </summary>
-        public IReadOnlyList<Drawable> HoveredDrawables => hoveredDrawables;
+        public IReadOnlyList<Drawable> HoveredDrawables => hoveredDrawables.AsSlimReadOnly();
```
Comment (smoogipoo) [inline osu.Framework/Input/InputManager.cs:110]:
```suggestion
        public SlimReadOnlyListWrapper<Drawable> HoveredDrawables => hoveredDrawables.AsSlimReadOnly();
```
That should fix the remaining allocation.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Framework/Input/InputManager.cs (hunk 1 above)
Comment (peppy) [reply osu.Framework/Input/InputManager.cs:110]: Hmm, right. Didn't realise we were happy with exposing these lists with the wrapper, but that *is* how it's done everywhere else.

**Review by smoogipoo — APPROVED** [review]

**Comment by peppy** [comment]
20210226 165004 (Parallels Desktop app)

**Pull request merged**